db/migrate/20191109022931_update_favorites_on_tracks.rb

```diff
@@ -0,0 +1,12 @@
+class UpdateFavoritesOnTracks < ActiveRecord::Migration[6.0]
+  def change
+    asset_ids = Track.favorites.map(&:asset_id).uniq
+    puts "Found #{asset_ids.length} favorite assets."
+
+    Asset.where(id: asset_ids).find_each do |asset|
+      if asset.favorites_count != asset.tracks.favorites.count
+        asset.favorites_count = asset.tracks.favorites.count
```

Comment:
I don't think this will actually update the database. We probably want to use `Asset.reset_counters` here https://api.rubyonrails.org/classes/ActiveRecord/CounterCache/ClassMethods.html#method-i-reset_counters

Reply:
Awesome.
Looks like `reset_counters` can be used on associations, while our favorites_count is a "custom" counter (I'm not sure that's how you classify it (we actually call `increment_` and `decrement_` counter to count those).
```
2.6.3 :006 > Asset.reset_counters(14, :favorites_count)
  Asset Load (1.0ms)  SELECT `assets`.* FROM `assets` WHERE `assets`.`deleted_at` IS NULL AND `assets`.`id` = 14 LIMIT 1
Traceback (most recent call last):
        2: from (irb):6
        1: from (irb):6:in `rescue in irb_binding'
ArgumentError ('Asset' has no association called 'favorites_count')
```
But we can use `update_counter` instead and pass it the value. Works great.

Reply:
The fact that it's a "custom counter" is what triggered my main PR comment about probably needing to move away from that towards normal rails counters.